### ISSUE TITLE
Support autoreplace for sbt manager

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#25643]
### Discussed in https://github.com/renovatebot/renovate/discussions/25637

<div type='discussions-op-text'>

<sup>Originally posted by **gaeljw** November  7, 2023</sup>
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Given the following repository: https://github.com/gaeljw/renovate-replacement-play, I'd like to get a PR opened using the "replacement" feature because one of my dependencies got its name changed.

- `"com.typesafe.play" % "sbt-plugin"` is renamed to `"org.playframework" % "sbt-plugin"`

### What you tried so far.

I've added the following replacement rules to my `renovate.json` config:

```json
"packageRules": [
      {
        "matchPackageNames": [
          "com.typesafe.play:sbt-plugin"
        ],
        "replacementName": "org.playframework:sbt-plugin",
        "replacementVersion": "3.0.0"
      }
    ]
```

But so far, I don't get any replacement PR opened. I only get new versions on the "old" naming, I'd like to get additional PRs with the new name and version 3.0.0.

### Relevant debug logs

<details><summary>Logs</summary>

```
"sbt": [
      {
        "deps": [
          {
            "datasource": "sbt-plugin",
            "depName": "com.typesafe.play:sbt-plugin",
            "packageName": "com.typesafe.play:sbt-plugin",
            "currentValue": "2.9.0",
            "depType": "plugin",
            "registryUrls": [
              "https://repo.maven.apache.org/maven2",
              "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases"
            ],
            "updates": [
              {
                "updateType": "replacement",
                "newName": "org.playframework:sbt-plugin",
                "newValue": "3.0.0",
                "branchName": "renovate/com.typesafe.play-sbt-plugin-replacement"
              }
            ],
            "versioning": "ivy",
            "warnings": [],
            "sourceUrl": "https://github.com/playframework/playframework",
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "dependencyUrl": "https://repo.maven.apache.org/maven2/com/typesafe/play/sbt-plugin",
            "currentVersion": "2.9.0",
            "fixedVersion": "2.9.0"
          }
        ],
        "packageFile": "project/plugins.sbt"
      }
    ]
  }

...

INFO: Cannot find replaceString in current file content. Was it already updated? (branch="renovate/com.typesafe.play-sbt-plugin-replacement")
{
  "packageFile": "project/plugins.sbt"
  "depName": "com.typesafe.play:sbt-plugin"
  "existingContent": "\naddSbtPlugin(\"com.typesafe.play\" % \"sbt-plugin\" % \"2.9.0\")\n"
  "replaceString": "2.9.0"
}

DEBUG: No content changed (branch="renovate/com.typesafe.play-sbt-plugin-replacement")
{
  "packageFile": "project/plugins.sbt"
  "depName": "com.typesafe.play:sbt-plugin"
}
```

</details>

I guess the error `Cannot find replaceString in current file content` is relevant but I'm not sure how to fix it.</div>

**Comment by gaeljw** [comment]
Happy to give a try at implementing this if someone can guide me in the right direction.

**Comment by rarkins** [comment]
In the meantime, the current docs should be updated to include SBT in the unsupported list: https://docs.renovatebot.com/configuration-options/#replacementname

**Comment by rarkins** [comment]
I searched the code for [replacementName](https://github.com/search?q=repo%3Arenovatebot%2Frenovate+replacementname&type=code) and unfortunately I'm not finding any _custom_ replacements. For example maven is not supported.

We may need to add some maven-awareness to the autoreplace function, e.g. instead of searching for the `packageName` string in full, we split it into group and package parts and then search for something like `groupPart.*?packagePart` and then replace them separately if found

**Comment by gaeljw** [comment]
Indeed, if this is not implemented for any other manager yet, this sounds like a bit of work.

**Comment by gaeljw** [comment]
Related to https://github.com/renovatebot/renovate/issues/14149

**Comment by MPV** [comment]
Should SBT also be listed here...?
- https://github.com/renovatebot/renovate/discussions/24883

**Comment by gaeljw** [comment]
I started a implementation in https://github.com/renovatebot/renovate/pull/39023 that happens to work but I'm really not sure it's the right way to go.

Now that replacement are supported in Maven, I think it's worth spending some time on making it work with sbt but from what I looked at the Maven implementation is very specific with XML handling, we cannot reuse it with sbt.

Help & feedback wanted :)
